cszczepaniak/cribbly-backend
# Add serilog

# What I added (link any issues addressed)
I added a logging framework called Serilog to allow us to have logs generated using rolling files. A good chunk of what I did was based on [IAmTimCorey's YouTube video](https://www.youtube.com/watch?v=_iryZxv8Rxw&t=1766s&ab_channel=IAmTimCorey) as well as the [official site](https://serilog.net/). 

I configured it to log every API request, and have also added logs in a few spots that I think would provide helpful info in a troubleshooting scenario. Obviously we can change whatever regarding what standards we want for log levels, when to log, etc. 

When running the application, it should now output our logs to the debug console, and append to .log files under a folder called "log" which you will create. Each day, a new file will be generated in the format of `BackendLog -- {yyyymmdd}`

# How to test it
Not sure what you look at as far as a debug console goes, but I found it helpful to add the following to my launch.json file in .vscode. This stops it from logging every file that gets loaded, so that only our logs will be displayed.

`"logging": {
      "moduleLoad": false
}`


- Restore the project
- Create a new folder inside CribblyBackend called "log"
- Run the project and do some stuff, see that a new .log file gets created and most of the endpoints should now show custom log messages

`[16:50:27 INF] HTTP GET /api/player/login responded 400 in 7.4552 ms`
`[16:51:23 DBG] New user created: {"Id": 11, "Email": "alex@rush.com", "Name": "Alex Lifeson", "Team": null, "Role": null, "$type": "Player"}`

As I was talking about, another thing we can do if we want is to output the logs into JSON, which will allow us to put it into visualization apps and whatnot. Might be overkill, but as far as logging goes, this should be a good start!
